File: public/templates/go/product.go
package main

import (
  "fmt"
  "net/http"
  "io"
)

func main() {
  resp, err := http.Get("https://real-time-backend-preview.vercel.app/api/product")
  if err != nil {
    fmt.Println("Error:", err)
    return
  }
  defer resp.Body.Close()
  body, err := io.ReadAll(resp.Body)
  if err != nil {
    fmt.Println("Error:", err)
    return
  }
  fmt.Println("Product:", string(body))
}

File: frontend/public/templates/typescript/product.ts
interface Product {
  id: number;
  name: string;
  price: string;
  description: string;
}

fetch('https://real-time-backend-preview.vercel.app/api/product')
  .then(response => response.json())
  .then((data: Product) => console.log('Product:', data))
  .catch((error: unknown) => console.error('Error:', error));